suitcase/pyxrf/export.py

```diff
@@ -0,0 +1,349 @@
+        Do not create data for each detector is data size is too large,
+        if set as false. This will slow down the speed of creating hdf file
+        with large data size. srx beamline only.
+    save_scalar : bool, optional
+        choose to save scaler data or not for srx beamline, test purpose only.
+    num_end_lines_excluded : int, optional
+        remove the last few bad lines
+    spectrum_len : int, optional
+        standard spectrum length
+    """
+
+    start_doc = hdr['start']
+    plan_n = start_doc.get('plan_name')
+    if 'fly' not in plan_n: # not fly scan, step scan instead
+        datashape = start_doc['shape']   # vertical first then horizontal
+        fly_type = None
+
+        snake_scan = start_doc.get('snaking')
+        if snake_scan[1] == True:
+            fly_type = 'pyramid'
+
+        try:
+            data = hdr.table(fill=True, convert_times=False)
+        except IndexError: # scan is not finished
+            total_len = get_total_scan_point(hdr) - 2
+            evs, _ = zip(*zip(hdr.events(hdr, fill=True), range(total_len)))
+            namelist = (config_data['xrf_detector'] +
+                        hdr.start.motors +config_data['scaler_list'])
+            dictv = {v:[] for v in namelist}
+            for e in evs:
+                for k,v in six.iteritems(dictv):
+                    dictv[k].append(e.data[k])
+            data = pd.DataFrame(dictv, index=np.arange(1, total_len+1)) # need to start with 1
+
+        xrf_detector_names = config_data['xrf_detector']
+        #express3 detector name changes in databroker
+        if xrf_detector_names[0] not in data.keys():
+            xrf_detector_names = ['xs_channel'+str(i) for i in range(1,4)]
+        logger.info('Saving data to hdf file.')
+        data_new = map_data2D(data, datashape,
+                              det_list=xrf_detector_names,
+                              pos_list=hdr.start.motors,
+                              fly_type=fly_type, spectrum_len=spectrum_len)
+        write_db_to_hdf_base(fpath, new_data, num_det=num_det,
+                             create_each_det=create_each_det)
+        if 'xs2' in hdr.start.detectors: # second dector
+            logger.info('Saving data to hdf file for second xspress3 detector.')
+            tmp = fpath.split('.')
+            fpath1 = '.'.join([tmp[0]+'_1', tmp[1]])
+            data_new = map_data2D(data, datashape,
+                                  det_list=config_data['xrf_detector2'],
+                                  pos_list=hdr.start.motors,
+                                  fly_type=fly_type, spectrum_len=spectrum_len)
+            write_db_to_hdf_base(fpath1, new_data, num_det=num_det,
+                                 create_each_det=create_each_det)
+    else:
+        # srx fly scan
+        # Added by AMK to allow flying of single element on xs2
+        if 'E_tomo' in start_doc['scaninfo']['type']:
+            num_det = 1
+
+        if save_scalar is True:
```

Comment:
The Python way to write this is `if save_scalar` so that any "truth-y" value will do. There are several cases like this below with the same issue.